packages/model-viewer/src/features/annotation.ts

```diff
@@ -0,0 +1,208 @@
+
+/* @license
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {property} from 'lit-element';
+import {Vector3} from 'three';
+import {CSS2DObject, CSS2DRenderer} from 'three/examples/jsm/renderers/CSS2DRenderer.js';
+import {_Math as ThreeMath} from 'three/src/math/Math.js';
+
+import {style} from '../decorators.js';
+import ModelViewerElementBase, {$onResize, $scene, $tick} from '../model-viewer-base.js';
+import {normalizeUnit} from '../styles/conversions.js';
+import {EvaluatedStyle, Intrinsics} from '../styles/evaluators.js';
+import {numberNode, NumberNode, parseExpressions} from '../styles/parsers.js';
+import {Constructor} from '../utilities.js';
+
+const DEFAULT_HIDDEN_OPACITY = 0.25;
+const DEFAULT_HIDDEN_ANGLE = Math.PI / 2;
+const TRANSITION_ANGLE = Math.PI / 8;
+
+const $annotationRenderer = Symbol('annotationRenderer');
+const $updateHotspots = Symbol('updateHotspots');
+const $hiddenAngle = Symbol('hiddenAngle');
+const $syncHiddenAngle = Symbol('syncHiddenAngle');
+const $nextHotspotIndex = Symbol('nextHotspotIndex');
+const $hotspotMap = Symbol('hotspotMap');
+const $observer = Symbol('observer');
+
+class Hotspot extends CSS2DObject {
+  public normal: Vector3;
+
+  constructor(element: HTMLElement) {
+    const wrapper = document.createElement('div');
+    const slot = document.createElement('slot');
+    slot.name = element.slot;
+    wrapper.appendChild(slot);
+    super(wrapper);
+    this.normal = new Vector3();
+    this.updatePosition(element);
+    this.updateNormal(element);
+  }
+
+  updatePosition(element: HTMLElement) {
+    const positionNodes = parseExpressions(element.dataset.position!)[0].terms;
+    for (let i = 0; i < 3; ++i) {
+      this.position.setComponent(
+          i, normalizeUnit(positionNodes[i] as NumberNode<'m'>).number);
+    }
+  }
+
+  updateNormal(element: HTMLElement) {
+    const normalNodes = parseExpressions(element.dataset.normal!)[0].terms;
+    for (let i = 0; i < 3; ++i) {
+      this.normal.setComponent(
+          i, normalizeUnit(normalNodes[i] as NumberNode<'m'>).number);
+    }
+  }
+}
+
+export declare interface AnnotationInterface {
+  hiddenOpacity: number;
```

Comment:
This seems like a good use case for a CSS custom property. Consider this [example](https://glitch.com/edit/#!/lopsided-motion?path=slotted-style-test.html) ([live version](http://lopsided-motion.glitch.me/slotted-style-test.html)).

With this approach, you would add or remove a class from the `<slot>`'s wrapper element depending on whether or not that hotspot is "hidden".